### DISCUSSION TITLE
[Firefox Android] Google search result URLs preserve tracking, unless Desktop site mode is used

### DISCUSSION THREAD

**Discussion by ThreeDeeJay** [uBlockOrigin/uAssets#20846] [Q&A]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is NOT an issue with [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

uBlock Origin successfully removes tracking from Google search result URLs for a direct link which allows loading certain links with the associated app, but that only works when enabling Desktop site mode.
This also happens when Enhanced tracking protection is completely disabled.

### A specific URL where the issue occurs.

`https://www.google.com/search?q=ublock+origin`

### Steps to Reproduce

- Google search anything on Firefox for Android with uBlock Origin. Example: `ublock origin` 
  `https://www.google.com/search?q=ublock+origin`
- In mobile mode, tap and hold a link to show it still has tracking like this: 

    `https://www.google.com/url?sa=AAAAA&source=web&rct=AAAAA&opi=AAAAA&url=https://ublockorigin.com/&ved=AAAAA=AAAAA`

https://github.com/uBlockOrigin/uBlock-issues/assets/71472458/37699362-c312-4728-977f-b8af4a1e0384

- Turn on Desktop site in the menu which refresh the page
- Tap and hold a search result again and this time it should show the direct URL like this: https://ublockorigin.com/

https://github.com/uBlockOrigin/uBlock-issues/assets/71472458/515d8cc4-4bb0-4bcc-b35c-da29c72a75f6

- Optionally, disable uBlock Origin then refresh to verify that without it, URLs have tracking even in Desktop site mode:
https://github.com/uBlockOrigin/uBlock-issues/assets/71472458/f7c0752a-54a1-4cde-80dd-87222cca31fa

### Expected behavior

uBlock origin should also sanitize/clean search result URLs in mobile (default) mode just like it does with Desktop mode.

### Actual behavior

Google search result URLs preserve their tracking parameters, which can prevent opening URLs in apps that have site associations and instead the links are opened in the same/new browser tab.

### uBO version

1.53.0

### Browser name and version

Firefox 120.0

### Operating System and version

Android 13

**Comment by MasterKia** [comment]
Duplicate of https://github.com/uBlockOrigin/uAssets/issues/7960

**Comment by ThreeDeeJay** [comment]
Ohh, my bad. I thought URL sanitation wasn't directly related to filter lists so I didn't check the uAssets repo.
Anyway, thanks for reopening the original 👍

**Comment by stephenhawk8054** [comment]
I think we can put it under `env_mobile`?

```adb
!#if env_mobile
www.google.com##+js(href-sanitizer, a[href^="/url?q=http"], ?q)
!#endif
```

**Reply by ThreeDeeJay** [reply]
Now that you mention it, I just checked and the issue doesn't seem to be Android device-specific.
Even on Windows, if I use User-Agent Switcher addon to use a Firefox Android agent, I can replicate it.
Also, with Android Phone/Tablet agents, the Chrome agent sanitizes the URL but Firefox's doesn't, but I'm not sure if this is a Firefox+mobile issue or something else like a different HTML structure 🤔 

I'm guessing the code above extracts the value of the `?q=` parameter so I'm not sure why it's not working with 
```https://www.google.com/url?q=https://ublockorigin.com/&sa=AAAAA&ved=AAAAA&usg=AAAAA```

On a side note, I also came across this format used by Windows+IE/Edge, which uses `url` instead of `q`:
`https://www.google.com/url?sa=AAAAA&source=web&rct=AAAAA&opi=AAAAA&url=https://ublockorigin.com/&ved=AAAAA-AAAAA&usg=AAAAA`

**Comment by ThreeDeeJay** [comment]
@mapx- I noticed you closed https://github.com/uBlockOrigin/uAssets/issues/7960#issuecomment-1825407116 as completed. Does that mean it was fixed in the development code or that it's just being continued here? I'm not familiar with Github Discussions so I'm not sure when to use instead of Issues 🤔

**Reply by mapx-** [reply]
just being continued here

**Comment by Animallearns** [comment]
The user asserts that the issue is not related to filter lists, YouTube, Facebook, or Twitch. They've ruled out support-related queries and checked for existing similar issues in the uBlock Origin issue tracker. The problem is specific to Google search result URLs when using uBlock Origin on Firefox for Android. The extension successfully removes tracking from the URLs when in Desktop site mode, but the issue persists in the default mobile mode, even with Enhanced tracking protection disabled. The user provides steps to reproduce the problem, including examples of URLs with and without tracking parameters. They expect uBlock Origin to sanitize search result URLs in both mobile and desktop modes but observe that the tracking parameters are preserved in mobile mode. The reported version of uBlock Origin is 1.53.0, and the browser is Fire fox 120.0 on Android 13.

This detailed report is intended to help the developers identify and address the issue within the uBlock Origin extension.

**Comment by krystian3w** [comment]
Maybe needed is add [Observer Mutation](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver (https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)) and copy url form second+ parameter's to `href-sanitizer`.

**Comment by areqamubeen** [comment]
may be helped for everyone

**Comment by areqamubeen** [comment]


**Reply by krystian3w** [reply]
Are you bot?

And what is it `https://leaflearns.com/plumeria-yellow/` hidden under dot...?

**Comment by areqamubeen** [comment]
> Are you bot?
> 
> And what is it `https://leaflearns.com/plumeria-yellow/` hidden under dot...?

no
